kv/range_cache.go

```diff
@@ -116,11 +121,10 @@ func (rmc *rangeDescriptorCache) LookupRangeDescriptor(key proto.Key,
 		log.Infof("lookup range descriptor: key=%s desc=%+v\n%s", key, r, rmc)
 	}
 
-	rs, err := rmc.db.getRangeDescriptor(key, options)
+	rs, err := rmc.db.getRangeDescriptors(key, options)
```

Comment:
I didn't think of this before, but there was likely merit to not locking completely over the retrieval of non-cached range descriptors. With this change, we're going to block everyone from using the cache while we're fetching anything, which we definitely don't want. In the previous version, you may launch multiple lookups in parallel which also isn't great, but doesn't completely block everyone who just wants to lookup a cached descriptor that has nothing to do with the update.
Ideally we'd have a keyrange-based waitgroup here for those calls which aren't cached (to make sure only the first such caller launches a call to the cluster) but that will be hard since the key ranges aren't necessarily known without the descriptors.